ssl/ktls.c

```diff
@@ -137,12 +137,15 @@ int ktls_check_supported_cipher(const SSL *s, const EVP_CIPHER *c,
           return 0;
 # endif
 # ifdef OPENSSL_KTLS_AES_GCM_128
+        /* fall thru */
```

Comment:
These comments should be more logically placed at the end of the cases that fall through.

Reply:
The ifdef is the problem. The comment has to be next to the case. If there is a ifdef/endif statement in between it does not count and the warning is generated.
An alternative would be to return 0 early so there is no need for  fall comment or an explicit break and an return at the end.

Reply:
IMO the compiler should catch the comment even if it is placed before the `# endif # ifdef`

Reply:
Hmm, you're right it seems gcc won't get it right if the comment is not right before the case.